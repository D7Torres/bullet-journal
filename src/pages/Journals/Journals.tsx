import { Link } from 'react-router-dom'

const currentYear = new Date().getFullYear()
const fromYear = currentYear - 5
const years = Array.from({ length: 10 }, (_, i) => fromYear + i)

export const Journals = () => (
  <>
    <h1>Journals</h1>
    <ul>
      {years.map((year) => (
        <li key={year}>
          <Link to={`/journal/${year}`}>{year}</Link>
        </li>
      ))}
    </ul>
  </>
)
