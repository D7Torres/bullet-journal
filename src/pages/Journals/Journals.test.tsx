import { expect, test } from 'vitest'
import { render } from '@testing-library/react'
import { createMemoryRouter, RouterProvider } from 'react-router-dom'
import { Journals } from './Journals'

test('Journals page renders', () => {
  const memoryRouter = createMemoryRouter(
    [
      {
        path: '/',
        element: <Journals />,
      },
    ],
    {
      initialEntries: ['/'],
    },
  )

  const { asFragment } = render(<RouterProvider router={memoryRouter} />)
  expect(asFragment()).toMatchSnapshot()
})
