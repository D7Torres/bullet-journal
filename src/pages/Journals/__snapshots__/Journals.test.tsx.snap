// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Journals page renders 1`] = `
<DocumentFragment>
  <h1>
    Journals
  </h1>
  <ul>
    <li>
      <a
        data-discover="true"
        href="/journal/2020"
      >
        2020
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2021"
      >
        2021
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2022"
      >
        2022
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2023"
      >
        2023
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2024"
      >
        2024
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2025"
      >
        2025
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2026"
      >
        2026
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2027"
      >
        2027
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2028"
      >
        2028
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/journal/2029"
      >
        2029
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
